File: n-ary-tree-level-order-traversal/n-ary-tree-level-order-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        def helper(root, depth, res):
            if not root: return []
            while len(res) <= depth: res.append([])
                
            res[depth].append(root.val) 
            for child in root.children:
                helper(child, depth+1, res)
        
        res = []
        helper(root, 0, res)
        return res
            

File: n-ary-tree-level-order-traversal/n-ary-tree-level-order-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> res;
        helper(root, 0, res);
        return res;
    }

public:
    void helper(Node* root, int depth, vector<vector<int>>& res){
        if (root == nullptr) return;
        
        while (res.size() <= depth) res.push_back({});
        res[depth].push_back(root->val);
        
        for (auto child : root->children){
            helper(child, depth+1, res);
        }
    }
};